keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/DeliveryArtifact.kt

```diff
@@ -96,6 +96,8 @@ abstract class DeliveryArtifact {
   @get:ExcludedFromDiff
   abstract val deliveryConfigName: String?
 
+  abstract fun copyWithName(name: String): DeliveryArtifact
```

Comment:
Consider renaming for accuracy:
```suggestion
  abstract fun withDeliveryConfigName(deliveryConfigName: String): DeliveryArtifact
```
Another alternative would be to just make `deliveryConfigName` a `var` instead of a `val`.